code/modules/multiz/movement.dm

```diff
@@ -208,29 +234,112 @@
 	if(locate(/obj/structure/stairs) in landing)
 		return 1
 
+	var/area/area1 = get_area(landing)
+	if(!area1.has_gravity())
+		return 1
+
+	if(istype(src,/mob/living) || isobj(src))
+		if(istype(landing, /turf/simulated/open))
+			z_levels_fallen++ // Don't damage them but keep track of this.
+			return 1
+
+		if(istype(landing, /turf/space))
+			z_levels_fallen = 0 // turns out they didn't hit anything solid. Lucky them.
+			return 1
+
+		if(!z_levels_fallen) // Checks if they only fell down one floor.
+			z_levels_fallen = 1
+
+		var/weight = 1
+		if(ismob(src))
+			var/mob/M = src
+			weight = M.mob_size
+		if(isobj(src))
+			var/obj/O = src
+			weight = O.w_class
+
+
+		var/fallforce = throw_range
```

Comment:
Non-objs not have a throwforce. Nob-objs still need a quantifiable damage check, so the next accurate variable is selected. It only comes into play for mobs and objs with a w_class over 2 but a throwforce less than 5.